sdmetrics/reports/single_table/_properties/column_pair_trends.py

```diff
@@ -184,7 +186,13 @@ def _get_columns_data_and_metric(
             data_real = real_data[[column_name_1, column_name_2]]
             data_synthetic = synthetic_data[[column_name_1, column_name_2]]
 
-        return data_real, data_synthetic, metric
+        metric_parameters = {}
+        if (metric == ContingencySimilarity) and (
+            max(len(data_real), len(data_synthetic)) > DEFAULT_NUM_ROWS_SUBSAMPLE
+        ):
+            metric_parameters['num_rows_subsample'] = DEFAULT_NUM_ROWS_SUBSAMPLE
+
+        return data_real, data_synthetic, metric, metric_parameters
```

Comment:
why not just do this in the calling method (_generate_details) instead of adding another return value to this method?